ibm/service/power/data_source_ibm_pi_instances.go

```diff
@@ -0,0 +1,284 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+)
+
+func DataSourceIBMPIInstances() *schema.Resource {
+
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIInstancesAllRead,
+		Schema: map[string]*schema.Schema{
+			helpers.PICloudInstanceId: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			// Computed Attributes
+			"pvm_instances": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"pvm_instance_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"memory": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+						"processors": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+						"health_status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"addresses": {
+							Type:       schema.TypeList,
+							Computed:   true,
+							Deprecated: "This field is deprecated, use networks instead",
```

Comment:
This is new data source, why adding a deprecated attribute. We can use networks directly,